tools/site_search.php

```diff
@@ -0,0 +1,343 @@
+}
+echo "</ul>";
+
+echo "<p>" . _("In addition, the following words can be used to jump to specific site pages.") . "</p>";
+
+echo "<ul>";
+foreach (get_jump_words() as $prefix => $data) {
+    echo "<li>";
+    echo "<kbd>$prefix</kbd> - " . $data["desc"];
+    if (@$data["alias"]) {
+        echo " (" . sprintf(_("alias: %s"), "<kbd>" . $data["alias"] . "</kbd>") . ")";
+    }
+    echo "</li>";
+}
+echo "</ul>";
+
+echo "<hr>";
+
+echo "<p>" . sprintf(
+    _("This page can be used with your browser's custom search feature (eg <a href='%s'>smart keywords</a> in Firefox, <a href='%s'>site shortcuts</a> in Chrome).<br>The URL to use in searching is <kbd>%s</kbd>"),
+    "https://support.mozilla.org/en-US/kb/how-search-from-address-bar",
+    "https://support.google.com/chrome/answer/95426",
+    "$code_url/tools/site_search.php?q=%s"
+) . "</p>";
+
+//----------------------------------------------------------------------------
+
+function get_prefix_searches(bool $flatten_aliases = false): array
+{
+    global $code_url, $wiki_url;
+
+    $prefix_searches = [
+        "title" => [
+            "url" => "$code_url/tools/search.php?show=search&title=%s",
+            "desc" => _("Project search for this in the title"),
+            "alias" => "t",
+        ],
+        "author" => [
+            "url" => "$code_url/tools/search.php?show=search&author=%s",
+            "desc" => _("Project search for this in the author"),
+            "alias" => "a",
+        ],
+        "language" => [
+            "url" => "$code_url/tools/search.php?show=search&language[]=%s",
+            "desc" => _("Project search for this language"),
+        ],
+        "pm" => [
+            "url" => "$code_url/tools/search.php?show=search&project_manager=%s",
+            "desc" => _("Project search for this project manager"),
+        ],
+        "user" => [
+            "url" => "$code_url/stats/members/mbr_list.php?uname=%s",
+            "desc" => _("Search for this username or part of a username"),
+            "alias" => "u",
+        ],
+        "team" => [
+            "url" => "$code_url/stats/teams/tlist.php?tname=%s",
+            "desc" => _("Search for this team name or part of a team name"),
+        ],
+        "task" => [
+            "url" => "$code_url/tasks.php?task_id=%s",
+            "desc" => _("Jump to this task ID"),
+            "validator" => "_validate_integer",
+        ],
+    ];
+
+    if (get_url_for_forum()) {
+        $prefix_searches["forum"] = [
+            "url" => get_url_for_search("%s", false),
+            "desc" => _("Search for this in forum posts"),
+        ];
+    }
+
+    if (!empty($wiki_url)) {
+        $prefix_searches["wiki"] = [
+            "url" => "$wiki_url/Special:Search/Index.php?search=%s",
+            "desc" => _("Search for this in the wiki"),
+        ];
+    }
+
+    if ($flatten_aliases) {
+        foreach ($prefix_searches as $key => $data) {
+            if (isset($prefix_searches[$key]["alias"])) {
```

Comment:
Isn't `$prefix_searches[$key]` just `$data` for the various occurrences this loop?

I tried it out and I think you can write this a bit more effectively as the follow (note the use of a reference in the foreach)

```
  foreach($prefix_searches as $key => &$data) {
      if (($alias = @$data["alias"]) !== null) {
          unset($data["alias"]);
          $prefix_searches[$alias] = $data;
      }
  }
```